x-pack/plugins/monitoring/server/routes/api/v1/elasticsearch/ccr.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Joi from 'joi';
+import { get } from 'lodash';
+import { handleError } from '../../../../lib/errors/handle_error';
+import { prefixIndexPattern } from '../../../../lib/ccs_utils';
+
+export function ccrRoute(server) {
+  server.route({
+    method: 'POST',
+    path: '/api/monitoring/v1/clusters/{clusterUuid}/elasticsearch/ccr',
+    config: {
+      validate: {
+        params: Joi.object({
+          clusterUuid: Joi.string().required()
+        }),
+        payload: Joi.object({
+          ccs: Joi.string().optional(),
+          timeRange: Joi.object({
+            min: Joi.date().required(),
+            max: Joi.date().required()
+          }).optional()
+        })
+      }
+    },
+    async handler(req, reply) {
+      const config = server.config();
+      const ccs = req.payload.ccs;
+      const esIndexPattern = prefixIndexPattern(config, 'xpack.monitoring.elasticsearch.index_pattern', ccs);
+
+      try {
+        const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('monitoring');
+        const response = await callWithRequest(req, 'search', {
+          index: esIndexPattern,
+          size: config.get('xpack.monitoring.max_bucket_size'),
+          filterPath: 'hits.hits._source.ccr_stats',
+          body: {
+            sort: [{ timestamp: { order: 'desc' } }],
+            query: {
+              bool: {
+                must: [
+                  {
+                    term: {
+                      type: {
+                        value: 'ccr_stats'
+                      }
+                    }
+                  }
+                ]
+              }
+            },
+            collapse: {
+              field: 'ccr_stats.follower_index'
+            },
+          }
+        });
+
+        const stats = get(response, 'hits.hits').map(hit => hit._source.ccr_stats);
+        const data = stats.reduce((accum, _stat) => {
+          const { allByFollowerIndex, shardStatsByFollowerIndex } = accum;
+
+          let follows = _stat.leader_index;
+          if (_stat.leader_index.includes(':')) {
+            const followsSplit = follows.split(':');
+            follows = `${followsSplit[1]} on ${followsSplit[0]}`;
```

Comment:
WDYT about storing this in a structured manner (i.e. an object with 2 keys) and letting the UI decide on the presentation ("SHARD on INDEX"). Not a huge deal as the server-side code and client-side code travel together, probably more of a nice-to-have.

Reply:
Yea I'd prefer it that way, but not a big deal either way.